tests/recipes/test_train_step.py

```diff
@@ -677,3 +694,62 @@ def _import_module(name: str, package: str = None):
 
         metrics = MlflowClient().get_run(run_id).data.metrics
         assert "training_f1_score" in metrics
+
+
+def test_train_step_with_predict_probability(
+    tmp_recipe_root_path: Path, tmp_recipe_exec_path: Path
+):
+    train_step_output_dir = setup_train_dataset(tmp_recipe_exec_path, recipe="classification")
+    recipe_steps_dir = tmp_recipe_root_path.joinpath("steps")
+    recipe_steps_dir.mkdir(parents=True)
+    recipe_yaml = tmp_recipe_root_path.joinpath(_RECIPE_CONFIG_FILE_NAME)
+    recipe_yaml.write_text(
+        """
+        recipe: "classification/v1"
+        target_col: "y"
+        primary_metric: "f1_score"
+        profile: "test_profile"
+        positive_class: "a"
+        run_args:
+            step: "train"
+        experiment:
+            name: "demo"
+            tracking_uri: {tracking_uri}
+        steps:
+            train:
+                using: custom
+                estimator_method: estimator_fn
+                tuning:
+                    enabled: false
+        """.format(
+            tracking_uri=mlflow.get_tracking_uri()
+        )
+    )
+    with mock.patch(
+        "steps.train.estimator_fn",
+        classifier_with_predict_proba_estimator_fn,
+    ):
+        recipe_config = read_yaml(tmp_recipe_root_path, _RECIPE_CONFIG_FILE_NAME)
+        train_step = TrainStep.from_recipe_config(recipe_config, str(tmp_recipe_root_path))
+        train_step.run(str(train_step_output_dir))
+
+    model_uri = get_step_output_path(
+        recipe_root_path=tmp_recipe_root_path,
+        step_name="train",
+        relative_path=TrainStep.MODEL_ARTIFACT_RELATIVE_PATH,
+    )
+    model = mlflow.pyfunc.load_model(model_uri)
```

Comment:
Looks like we don't have any test coverage verifying that the model can be loaded with `mlflow.sklearn.load_model()`. Can we add some?